extensions/cl_exp_tensor.asciidoc

```diff
@@ -0,0 +1,1058 @@
+  NULL. This extension does not define any optional properties for
+  tensors, but future extensions may define properties.
+
+* _shape_ defines the extends of the tensor's dimensions in number of
+  elements.
+
+* _layout_ points to an optional structure describing how tensor
+  elements are laid out in the buffer memory. The structure must be a
+  type corresponding to the _layout_type_ listed in
+  <<layout-types-table, tensor layout type>> table. The pointer is
+  ignored if the _tensor_type_ is `CL_TENSOR_LAYOUT_OPAQUE_EXP`.
+
+* _layout_type_ indicates the layout structure type the _layout_
+  point to.
+
+
+[[tensor-dtypes-table]]
+.Tensor element types. The API type indicates the corresponding type for copying elements from an host allocation / buffer object to tensor or vice versa.
+[cols="1,1,1",stripes=even]
+|===
+| *Tensor element data type* | *Description* | *API type*
+
+| CL_TENSOR_DTYPE_BOOL | Data type representing true or false.  |
+cl_uchar. footnote:[zero and non-zero bytes are interpreted as false
+and true values, respectively.]
+
+| CL_TENSOR_DTYPE_INT4_EXP        | 4-bit signed integer.            | cl_char.
+| CL_TENSOR_DTYPE_INT8_EXP        | 8-bit signed integer.            | cl_char.
+| CL_TENSOR_DTYPE_INT16_EXP       | 16-bit signed integer.           | cl_short.
+| CL_TENSOR_DTYPE_INT32_EXP       | 32-bit signed integer.           | cl_int.
+| CL_TENSOR_DTYPE_INT64_EXP       | 64-bit signed integer.           | cl_long.
+| CL_TENSOR_DTYPE_UINT8_EXP       | 8-bit unsigned integer.          | cl_uchar.
+| CL_TENSOR_DTYPE_UINT16_EXP      | 16-bit unsigned integer.         | cl_ushort.
+| CL_TENSOR_DTYPE_UINT32_EXP      | 32-bit unsigned integer.         | cl_uint.
+| CL_TENSOR_DTYPE_UINT64_EXP      | 64-bit unsigned integer.         | cl_ulong.
+| CL_TENSOR_DTYPE_FP8_EXP         | Half precision floating-point.   | cl_char.
+| CL_TENSOR_DTYPE_FP16_EXP        | Half precision floating-point.   | cl_half.
+| CL_TENSOR_DTYPE_BFLOAT16_EXP    | 16-bit brain floating-point.     | cl_ushort
+| CL_TENSOR_DTYPE_FP32_EXP        | Single precision floating-point. | cl_float.
+| CL_TENSOR_DTYPE_FP64_EXP        | Double precision floating-point. | cl_double.
+| CL_TENSOR_DTYPE_COMPLEX64_EXP   | 64-bit complex floating-point with
+  32-bit real and imaginary part. | cl_float2
+| CL_TENSOR_DTYPE_COMPLEX128_EXP  | 128-bit complex floating-point with
+  64-bit real and imaginary part. | cl_double2
+|===
+
+[[layout-types-table]]
+.Optional tensor memory layout types.
+[cols="1,1,4",stripes=even]
+|===
+| *layout type* | *tensor layout type* | *Description*
+
+| CL_TENSOR_LAYOUT_OPAQUE_EXP | N/A | The tensor don't have application
+  defined memory layout. Driver controls the tensors layout. To read
+  or write elements of the tensor
+
+| CL_TENSOR_LAYOUT_BLAS_EXP
+|<<cl-tensor-layout-blas,cl_tensor_layout_blas_exp>>
+| A type that describe packed memory layout similar ones used in BLAS APIs.
```

Comment:
```suggestion
| A type that describes a packed memory layout similar ones used in BLAS APIs.
```

Reply:
Is this accurate enough description. Is the BLAS layout well-specified somewhere to which we can refer?